demos/rust-cosmos-broadcaster/src/service.rs

```
@@ -0,0 +1,69 @@
+use crate::DEFAULT_VALIDATOR_RPC;
+use bs58;
+use cosmrs::rpc::{Client, HttpClient};
+use cosmrs::{tendermint, AccountId};
+use nym_validator_client::nyxd::CosmWasmClient;
+
+pub async fn create_broadcaster() -> HttpClient {
+    let broadcaster: HttpClient = HttpClient::new(DEFAULT_VALIDATOR_RPC).unwrap();
+    broadcaster
+}
+
+pub async fn get_sequence(
+    broadcaster: HttpClient,
+    signer_address: AccountId,
+) -> Result<crate::SequenceRequestResponse, std::io::Error> {
+    // get signer information
+    let sequence = broadcaster.get_sequence(&signer_address).await.unwrap();
```

if we're already returning a `Result` here, why not propagate the error?

---

you might also need to cover a special case when the account does not have a signing sequence (which will be the case for any fresh account)